### DISCUSSION TITLE
favicon / sitemeta question

### DISCUSSION THREAD

**Discussion by wolfspyre** [jpanther/congo#713] [Q&A] [answered]
related to https://github.com/jpanther/congo/discussions/711 /  https://github.com/jpanther/congo/discussions/683

in `layouts/partials/head.html:93-100` 
you add metadata linking to favicon assets and site.webmanifest from pages.

however the assets in question currently reside in /static; and so regardless of if the user wishes them to be or not,  these files are added to a site unless explicitly overwritten.

I don't think this is the intended behavior. 
I THINK the intent is to have these files reside in /assets, and be deployed as a fallback in the event that the user doesn't have something else configured.

am I missing something? or is that correct? 

if it's correct, I can file a PR relocating them from static to assets, unless you have something else in mind?

**Comment by jpanther** [comment]
I think they just ended up in static because way back when I created the project that's where I had them. Moving things to assets has been a very gradual process and this is one of the last remnants. They haven't really been an urgent thing for me to move because favicons are the definition of a static asset and I don't know what the benefit would be by putting them into assets?

**Reply by wolfspyre** [reply]
the benefit is:
- congo creates a few static assets with names of its’ choosing (these names are not unreasonable, but they’re not ubiquitously agreed upon conventionally…. because standards are wobly)
  - favicon.ico
  - favicon-16x16.png
  - favicon-32x32.png
  - apple-touch-icon.png
  - android-chrome-192x192.png
  - android-chrome-512x512.png
  - site.webmanifest

CURRENTLY, the ONLY way to prevent those assets from becoming part of a site is to redeclare them. 

being specific, take site.webmanifest.
the naming of a webmanifest asset is wobly. 
it could be 
manifest.webmanifest, manifest.json, site.json or site.webmanifest… plus a few other variants.

the existence of the site.webmanifest file in static means that if a user creates a manifest.webmanifest file, that file will be deployed adjacent congo’s site.webmanifest resource

even if the file ‘site.webmanifest’ is not explicitly mentioned in html, some web tools will “helpfully” try to “find the needful” preemptively, but get incorrect data due to the site.webmanifest resource existing in the webroot, but being outside the control of the user.


I’m currently creating a webmanifest asset dynamically; which gets laid down as manifest.webmanifest

this wouldn’t be a problem if the site.webmanifest file and related files lived in assets/, as creating a custom favicons.html partial would bypass the declaration logic that adds these assets to site html files and publishes them… 

however, as-is; the mere existence of them in the webroot is kinda problematic as i cannot easily copy the already-generated manifest to an additional location. (overwriting site.webmanifest)

so, i have to create another copy of the webmanifest 
 dynamically which must  reference all the same favicon assets i declared for the first webmanifest asset, 
but not attempt to re-create the images/icons/assets
(because then ill have two sources declaring the same child image resource and hugo will get crankypants™️)

so… yer not wrong that it’s not a HUGE problem, but it does introduce an unnecessary hurdle.

does that make sense?

**Reply by jpanther** [reply]
Yeah okay, it does make sense and it probably has caused issues for people and it has never been reported. I have no idea what the standards are around web manifests either. The current version and file naming is all simply because that's what the favicon generator I used output. Favicons used to be so simple.. now it's just ridiculous how many variants are needed.

**Reply by wolfspyre** [reply]
RIGHT THERE WITH YOU.

Why this all got so damned complex/complicated is.... well....  more likely just as much related to politics as anything else... 

- android/apple/microsoft each optimizing for themselves
- myriad competing "standards"
- incomplete information on the internet
- misinformation on the internet

Like I'd mentioned in #682 this is a pretty astoundingly thorough writeup on the state of the landscape and is pretty recent:

https://dev.to/masakudamatsu/favicon-nightmare-how-to-maintain-sanity-3al7

**Reply by jpanther** [reply, marked as answer]
I've created #720 to bundle all these site manifest changes together.
